concent_api/core/subtask_helpers.py

```diff
@@ -311,7 +313,12 @@ def finalize_deposit_claim(
     )
 
     if deposit_claim:
-        bankster.finalize_payment(deposit_claim)
+        tx_hash = bankster.finalize_payment(deposit_claim)
+        payment_interface = PaymentInterface()  # new object is NOT created, it is singleton and its instance is returned
+        payment_interface.on_transaction_confirmed(   # type: ignore  # pylint: disable=no-member
+            tx_hash=tx_hash,
+            cb=lambda: discard_claim(deposit_claim)
+        )
```

Comment:
Please make sure that `discard_claim()` does not fail if two processes try to do it at the same time (i.e. first gets a lock, seconds waits on the lock, first runs `delete()` and releases the lock, second runs `delete()` on the same, already deleted object). I think it won't fail on the SQL level (the `WHERE` clause in the second `DELETE` will simply match no rows) but I'm not sure how Django will behave.